packages/digit_dss/lib/pages/dashboard/dashboard.dart

```diff
@@ -0,0 +1,158 @@
+
+import '../../utils/i18_key_constants.dart' as i18;
+import '../../utils/utils.dart';
+import '../../widgets/dashboard/dashboard_metric_card.dart';
+import '../../widgets/localized.dart';
+
+@RoutePage()
+class UserDashboardPage extends LocalizedStatefulWidget {
+  const UserDashboardPage({
+    super.key,
+  });
+
+  @override
+  State<StatefulWidget> createState() {
+    return UserDashboardPageState();
+  }
+}
+
+class UserDashboardPageState extends LocalizedState<UserDashboardPage> {
+  @override
+  void initState() {
+    context.read<DashboardBloc>().add(DashboardRefreshEvent(
+          projectId: DashboardSingleton().projectId,
+          syncFromServer: false,
+          selectedDate: DateTime.now(),
+        ));
+
+    super.initState();
+  }
+
+  bool isLoading = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocConsumer<DashboardBloc, DashboardState>(
+        listener: (context, dashboardState) {
+      dashboardState.maybeWhen(
+          orElse: () => false,
+          loading: () {
+            if (!isLoading) {
+              setState(() {
+                isLoading = true;
+              });
+              Loaders.showLoadingDialog(context);
+            }
+          },
+          fetched: (metricData, tableData, selectedDate) {
+            Navigator.of(context, rootNavigator: true).pop();
+            setState(() {
+              isLoading = false;
+            });
+          },
+          error: () {
+            Navigator.of(context, rootNavigator: true).pop();
+            setState(() {
+              isLoading = false;
+            });
+            DigitToast.show(context,
+                options: DigitToastOptions(
+                    localizations.translate(i18.dashboard.someErrorOccured),
+                    true,
+                    DigitTheme.instance.mobileTheme));
+          });
+    }, builder: (context, dashboardState) {
+      return RefreshIndicator(
+        onRefresh: () {
+          dashboardState.maybeWhen(
+              orElse: () => false,
+              fetched: (metricData, tableData, selectedDate) async {
+                bool isConnected = await getIsConnected();
+                if (isConnected) {
+                  context.read<DashboardBloc>().add(DashboardRefreshEvent(
+                        projectId: DashboardSingleton().projectId,
+                        syncFromServer: true,
+                        selectedDate: selectedDate ?? DateTime.now(),
+                      ));
+                } else {
+                  DigitToast.show(context,
+                      options: DigitToastOptions(
+                        localizations
+                            .translate(i18.dashboard.networkFailureError),
+                        true,
+                        DigitTheme.instance.mobileTheme,
+                      ));
+                }
+              });
+
+          return Future<void>.delayed(const Duration(seconds: 1));
```

Comment:
Remove the delay and create a method of future return

Reply:
This is the required return type for onRefresh function, Need to specify the time for which the refresh indicator will be shown